pelican/plugins/seo/tests/test_seo_report.py

```diff
@@ -52,7 +52,66 @@ def test_generate_create_report_file_and_write_output(
     ):
         """
         Test that generate create a HTML file and write SEO report on it.
-        Need mock_open to test this.
+        """
+
+        output = self._generate_seo_report(fake_seo_report, fake_articles_analysis)
+        assert "<h1>SEO report - Fake site</h1>" in output
+
+    def test_report_contains_correct_analysis(
+        self, fake_seo_report, fake_articles_analysis
+    ):
+        """
+        Test that generated HTML SEO report contains expected analysis.
+        """
+
+        output = self._generate_seo_report(fake_seo_report, fake_articles_analysis)
+
+        # title
+        # 3 out of 3 articles have included a title
+        assert output.count("You have declared a title. Nice job!") == 3
+        # 1 out of 3 articles has a correct title
+        assert output.count("Your title has a good length.") == 1
+        # 2 out of 3 articles have a title that is too short
+        assert (
+            output.count(
+                "Your title is too short. The recommended length is 70 characters."
+            )
+            == 2
+        )
+
+        # description
+        # 2 out of 3 articles have included a description
+        assert output.count("You have declared a description. Nice job!") == 2
+        # 1 out of 3 articles has a correct description
+        assert output.count("Your description has a good length.") == 1
+        # 1 out of 3 articles have a title that is too short
+        assert (
+            output.count(
+                "Your description is too short. The minimum recommended length is 150 characters."
+            )
+            == 1
+        )
+        # 1 out of 3 articles is missing a description
+        assert output.count("You need to declare a description to improve SEO.") == 1
+
+        # content title
+        # 2 out of 3 articles have included a content title
+        assert output.count("You have declared a content title. Nice job!") == 2
+        # 1 out of 3 articles has a non-unique content title
+        assert output.count("Your content title must be unique.") == 1
+        # 1 out of 3 articles is missing the content title
+        assert output.count("You're missing a content title.") == 1
+
+        # internal links
+        # 2 out of 3 articles have 2 internal links
+        assert output.count("You've included 2 internal links. Nice job!") == 2
+        # 1 out of 3 articles is missing internal links
+        assert output.count("It's better to include internal links.") == 1
+
+    def _generate_seo_report(self, fake_seo_report, fake_articles_analysis):
```

Comment:
I would go for a pytest.fixture in this case (just after the import part), called for example `seo_report`. Thus, you'll be able to call the two tests with something like this `def my_test(seo_report)`.

Otherwise, the rest is good to me :)